packages/core/src/service/services/asset.service.ts

```diff
@@ -239,6 +267,78 @@ export class AssetService {
                 }),
             };
         }
+        const hasDeleteAllPermission = await this.hasDeletePermissionForChannels(ctx, channelsOfAssets);
+        if (deleteFromAllChannels && !hasDeleteAllPermission) {
+            throw new ForbiddenError();
+        }
+        if (!deleteFromAllChannels) {
+            assets.map(asset =>
+                this.channelService.removeFromChannels(ctx, Asset, asset.id, [ctx.channelId]),
```

Comment:
Should be be publishing the `AssetChannelEvent` 'removed' here?

Reply:
Yes, and I also forgot to `Promise.all()` for both async `removeFromChannels`